pkg/csi/service/common/authmanager.go

```diff
@@ -221,27 +225,35 @@ func GenerateDatastoreMapForFileVolumes(ctx context.Context, vc *vsphere.Virtual
 	// Return empty map if no vSAN datastores are found.
 	if len(vsanDsURLToInfoMap) == 0 {
 		log.Debug("No vSAN datastores found")
-		return dsURLToInfoMap, nil
+		return clusterToDsInfoListMap, nil
 	}
-	var allvsanDatastoreUrls []string
-	for dsURL := range vsanDsURLToInfoMap {
-		allvsanDatastoreUrls = append(allvsanDatastoreUrls, dsURL)
+
+	// Initialize vsan client
+	err = vc.ConnectVsan(ctx)
+	if err != nil {
+		log.Errorf("error occurred while connecting to VSAN, err: %+v", err)
+		return nil, err
 	}
-	fsEnabledMap, err := IsFileServiceEnabled(ctx, allvsanDatastoreUrls, vc)
+
+	fsEnabledClusterToDsURLsMap, err := getFSEnabledClusterToDsURLsMap(ctx, vc)
 	if err != nil {
-		log.Errorf("failed to get if file service is enabled on vsan datastores with error %+v", err)
+		log.Errorf("failed to get if file service enabled clusters map with error %+v", err)
```

Comment:
log.Errorf("failed to get if file service enabled clusters map with error %+v", err) ==> log.Errorf("failed to get file service enabled clusters map with error %+v", err)